docs/USAGE-GUIDE.md

```diff
@@ -1585,6 +1585,102 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *n
+int s2n_offered_psk_get_identity(struct s2n_offered_psk *psk, uint8_t** identity, uint16_t *size);
+bool s2n_offered_psk_list_has_next(struct s2n_offered_psk_list *psk_list);
+int s2n_offered_psk_list_next(struct s2n_offered_psk_list *psk_list, struct s2n_offered_psk *psk);
+int s2n_offered_psk_list_reread(struct s2n_offered_psk_list *psk_list);
+int s2n_offered_psk_list_choose_psk(struct s2n_offered_psk_list *psk_list, struct s2n_offered_psk *psk);
+typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn, void *context,
+                                          struct s2n_offered_psk_list *psk_list);
+int s2n_config_set_psk_selection_callback(struct s2n_config *config, s2n_psk_selection_callback cb, void *context);
+```
+
+### Configuring External Pre-Shared Keys
+
+Use the following APIs to configure and use external pre-shared keys in the > =TLS1.3 connections. An external PSK is an out-of-band provisioned pre-shared key (PSK) established externally using a secure mutually agreed upon mechanism.
+
+To create an external PSK a caller must call the `s2n_external_psk_new` API.**s2n_external_psk_new** creates a new s2n external pre-shared key (PSK) object with `S2N_PSK_HMAC_SHA256` as the default pre-shared key hash algorithm. Note that the obfuscated ticket age will always be zero for external PSKs. Use `s2n_psk_free` to free the memory allocated to the s2n external pre-shared key object created by this API.
+
+**s2n_psk_free** frees the memory associated with the `s2n_psk` object.
+
+After creation of the external PSK object use `s2n_psk_set_identity`, `s2n_psk_set_secret` to set the PSK Identity and PSK secret. To override the default PSK Hash Algorithm  use `s2n_psk_set_hmac`.
+
+**s2n_psk_set_identity** sets the pre-shared key (PSK) identity for a given s2n_psk object. The pre-shared key identity is a unique identifier for the pre-shared secret. The Pre-shared key identity is transmitted over the network unencrypted and is a non-secret value. Be sure to not include any confidential information as a part of the pre-shared key identity. Note that the identity value is copied into s2n-tls memory and the user is responsbile for freeing the memory  associated with the `identity` input. Be sure to call `free(identity)` after this API call. 
+
+**s2n_psk_set_secret** sets the out-of-band/externally provisioned pre-shared secret (PSK) for a given s2n_psk object. Note that the secret value is copied into s2n-tls memory and the user is responsbile for freeing the memory associated with the `secret` input. Be sure to call `free(secret)` after this API call.
+
+**s2n_psk_set_hmac** sets the PSK HMAC algorithm for a given PSK object.
+The supported pre-shared key HMAC algorithms are listed in the `s2n_psk_hmac` enum. Note that in versions >= TLS1.3 the cipher suite selected has the same Hash alogrithm as the pre-shared key. Use this API to override the default PSK Hash Algorithm value `S2N_PSK_HMAC_SHA256`. 
```

Comment:
What is the HMAC for? When should a customer call this API?

Reply:
The last line already calls that out:`Use this API to override the default PSK Hash Algorithm value `S2N_PSK_HMAC_SHA256`.

Reply:
That tells me what the API is for, but not what the HMAC algorithm is for. When would I want to override the HMAC?